Assets/Scripts/Player/TrackHuskyDollars.cs

```diff
@@ -0,0 +1,67 @@
+using System.Collections;
+using System.Collections.Generic;
+using Unity.VisualScripting;
+using UnityEditor.SearchService;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class TrackHuskyDollars : MonoBehaviour
+{
+    public int startingBalance = 100;
+
+    public static Text balanceText;
+    public static Text balanceTextShadow;
+
+    public AudioClip moneySound;
+    private static AudioClip moneySoundStatic;
+
+    private static int currentBalance;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        currentBalance = startingBalance;
+        moneySoundStatic = moneySound;
+
+        GameObject[] objects = GameObject.FindGameObjectsWithTag("MoneyUI");
+        balanceText = objects[0].GetComponent<Text>();
+        balanceTextShadow = objects[1].GetComponent<Text>();
+        UpdateBalanceText();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
+
+    void OnCollisionEnter(Collision other) {
+        if (other.gameObject.CompareTag("Money")) {
+            AddHuskyDollars(10);
+            Destroy(other.gameObject);
+        }
+    }
+
+    public static void AddHuskyDollars(int amount) {
+        if (moneySoundStatic)
+        {
+            AudioSource.PlayClipAtPoint(moneySoundStatic, Camera.main.transform.position);
+        }
+        currentBalance += amount;
+        UpdateBalanceText();
+    }
+
+    public static void SubtractHuskyDollars(int amount) {
+        currentBalance -= amount;
+        UpdateBalanceText();
+    }
+
+    public static int GetCurrentBalance() {
+        return currentBalance;
+    }
+
+    private static void UpdateBalanceText() {
+        balanceText.text = "$" + currentBalance + ".00";
```

Comment:
Let's remove the decimals because we won't ever have decimal money

Reply:
was more for the look, the UI looked nice having the decimals. removed it though